protocol/messenger_communities.go

```diff
@@ -2720,24 +2732,44 @@ func (m *Messenger) requestCommunitiesFromMailserver(communities []communities.C
 			//we don't remember filter id associated with community because it was already installed
 			m.requestedCommunities[c.CommunityID] = nil
 		}
-		topics = append(topics, filter.ContentTopic)
+
+		groupedTopics[filter.PubsubTopic] = append(groupedTopics[filter.PubsubTopic], filter.ContentTopic)
 	}
 
+	defer func() {
+		for _, c := range communities {
+			m.forgetCommunityRequest(c.CommunityID)
+		}
+	}()
+
 	to := uint32(m.transport.GetCurrentTime() / 1000)
 	from := to - oneMonthInSeconds
 
-	_, err := m.performMailserverRequest(func() (*MessengerResponse, error) {
-		batch := MailserverBatch{From: from, To: to, Topics: topics}
-		m.logger.Info("Requesting historic")
-		err := m.processMailserverBatch(batch)
-		return nil, err
-	})
-
-	if err != nil {
-		m.logger.Error("Err performing mailserver request", zap.Error(err))
-		return
+	wg := sync.WaitGroup{}
+
+	for pubsubTopic, contentTopics := range groupedTopics {
+		wg.Add(1)
+		go func(pubsubTopic string, contentTopics []types.TopicType) {
+			_, err := m.performMailserverRequest(func() (*MessengerResponse, error) {
+				batch := MailserverBatch{
+					From:        from,
+					To:          to,
+					Topics:      contentTopics,
+					PubsubTopic: pubsubTopic,
+				}
+				m.logger.Info("requesting historic", zap.Any("batch", batch))
```

Comment:
Consider logging the communityID here too:
```suggestion
				m.logger.Info("requesting historic", zap.Any("batch", batch), zap.String("communityID", communityID))
```

Reply:
hmm, we don't really have `communityID` at this stage 🤔

Reply:
nvm then! :)